test/unittests/xmlupload/test_list_client_live.py

```diff
@@ -0,0 +1,244 @@
+                    {
+                        "id": "http://www.example.org/lists#a1",
+                        "name": "node-a1",
+                    },
+                    {
+                        "id": "http://www.example.org/lists#a2",
+                        "name": "node-a2",
+                    },
+                ],
+            }
+        }
+        con = ConnectionMock([list_iris, list_a])
+        list_client = ListClientLive(con, "")
+        lookup = list_client.get_list_node_id_to_iri_lookup()
+        expected = {
+            "list-a:list-a": "http://www.example.org/lists#a",
+            "list-a:node-a1": "http://www.example.org/lists#a1",
+            "list-a:node-a2": "http://www.example.org/lists#a2",
+        }
+        assert lookup == expected
+
+    def test_multiple_lists(self) -> None:
+        list_iris = {
+            "lists": [
+                {"id": "http://www.example.org/lists#a"},
+                {"id": "http://www.example.org/lists#b"},
+            ]
+        }
+        list_a = {
+            "list": {
+                "listinfo": {
+                    "id": "http://www.example.org/lists#a",
+                    "name": "list-a",
+                },
+                "children": [
+                    {
+                        "id": "http://www.example.org/lists#a1",
+                        "name": "node-a1",
+                    },
+                ],
+            }
+        }
+        list_b = {
+            "list": {
+                "listinfo": {
+                    "id": "http://www.example.org/lists#b",
+                    "name": "list-b",
+                },
+                "children": [
+                    {
+                        "id": "http://www.example.org/lists#b1",
+                        "name": "node-b1",
+                    },
+                ],
+            }
+        }
+        con = ConnectionMock([list_iris, list_a, list_b])
+        list_client = ListClientLive(con, "")
+        lookup = list_client.get_list_node_id_to_iri_lookup()
+        expected = {
+            "list-a:list-a": "http://www.example.org/lists#a",
+            "list-a:node-a1": "http://www.example.org/lists#a1",
+            "list-b:list-b": "http://www.example.org/lists#b",
+            "list-b:node-b1": "http://www.example.org/lists#b1",
+        }
+        assert lookup == expected
+
+
+class TestGetListFromServer:
+    """Test the _get_list_info_from_server function."""
+
+    def test_no_name_no_children(self) -> None:
+        list_response = {
+            "list": {
+                "listinfo": {
+                    "id": "http://www.example.org/lists#a",
+                    "labels": [{"value": "some label"}],
+                },
+                "children": [],
+            }
+        }
+        con = ConnectionMock([list_response])
+        result = _get_list_from_server(con, "")
+        expected_node = ListNode(node_iri="http://www.example.org/lists#a", node_name="some label")
```

Comment:
I'm confused that "some label" is the **name** of the node. According to line 129, I expect it to be the **label**. Perhaps you could improve naming in line 129? (on that line, I also don't get what "value" is)

Reply:
no, I can't change that, the JSON you're seeing is of the same shape as the JSON returned by the list endpoint of the API (just with lots of non-important information removed). And because the list at hand here, does not have a `name` specified, the label will be used as a name instead. (Which is not ideal, but the problem is not that in and by itself, but that we can even have list nodes without a unique name.)